File: api/models/announcements.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class AnnouncementBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    message: str = Field(..., min_length=1)


class AnnouncementCreate(AnnouncementBase):
    pass


class AnnouncementUpdate(AnnouncementBase):
    pass


class Announcement(AnnouncementBase):
    id: int
    author_id: int
    author_name: str
    author_email: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class AnnouncementListItem(BaseModel):
    id: int
    title: str
    message: str
    author_id: int
    author_name: str
    created_at: datetime
    updated_at: datetime

File: app/types/announcements.ts
export interface Announcement {
  id: number;
  title: string;
  message: string;
  author_id: number;
  author_name: string;
  author_email?: string;
  created_at: string;
  updated_at: string;
}

export interface AnnouncementCreate {
  title: string;
  message: string;
}

export interface AnnouncementUpdate {
  title: string;
  message: string;
}

